Sprint 12

DarkRavenn opened the pull request.

DarkRavenn committed:
временный комит

DarkRavenn committed:
Merge branch 'main' of github.com:DarkRavenn/ImageFeed into sprint_12

DarkRavenn committed:
- Добавил вывод в консоль при прокрутке до последней ячейки
- добавил мок массив photos: [Photo]

DarkRavenn committed:
Добавлена загрузка и отображение фото с сервера

DarkRavenn committed:
Добавлен сетевой слой и обработка ответа сервера для лайков

DarkRavenn committed:
Функционал лайков переделан в паттерн делегат

DarkRavenn committed:
Исправлено полноэкранное отображение фото, поправил код стайл

DarkRavenn committed:
Добавлен функционал выхода из профиля

DarkRavenn committed:
Добавил отображение плейсхолдера если не удалось загрузить большое фото

andrew-kuzn-yndx commented on ImageFeed/ImagesList/ImagesListCell.swift:
```diff
@@ -6,11 +6,42 @@
+    // MARK: - IB Outlets
     @IBOutlet var cellImage: UIImageView!
     @IBOutlet var likeButton: UIButton!
     @IBOutlet var dateLabel: UILabel!
```
Можно сделать аутлеты ImagesListCell как private, и менять их внутри ячейки.
Я понимаю, что ты работаешь с этими свойствами из контроллера, но работать напрямую с аутлетами из другого класса - не лучшая практика.
Лучше сделать публичную функцию, например configCell с аргументами на вход, и уже в ней менять аутлеты.
Но это, конечно же, просто идея на будущее)

andrew-kuzn-yndx commented on ImageFeed/ImagesList/ImagesListCell.swift:
```diff
@@ -6,11 +6,42 @@
+    }
+    
+    // MARK: - IB Actions
+    @IBAction func likeButtonTap(_ sender: Any) {
```
Не забывай, что IBAction всегда должны быть приватными)

andrew-kuzn-yndx commented on ImageFeed/ImagesList/ImagesListService.swift:
```diff
@@ -0,0 +1,209 @@
+    
+    private func addToPhotos(newPhotos: [PhotoResult]) {
+        let dateFormater = DateFormatter()
+        dateFormater.dateFormat = "yyyy-MM-dd'T'HH:mm:ss'Z'"
```
Давай DateFormatter сделаем свойством класса, чтобы не создавать каждый раз внутри метода
А еще лучше выделить его создание в отдельный глобальный extension, чтобы создавать его один раз для всего проекта

andrew-kuzn-yndx commented on ImageFeed/ImagesList/ImagesListViewController.swift:
```diff
@@ -57,32 +89,38 @@ extension ImagesListViewController: UITableViewDataSource {
 
+// MARK: - Extansion
 extension ImagesListViewController {
     func configCell(for cell: ImagesListCell, with indexPatch: IndexPath) { 
```
Вот этот метод можно почти полностью перенести в ячейку)

andrew-kuzn-yndx commented on ImageFeed/ImagesList/ImagesListViewController.swift:
```diff
@@ -57,32 +89,38 @@ extension ImagesListViewController: UITableViewDataSource {
+        cell.cellImage.kf.indicatorType = .activity
+        cell.cellImage.kf.setImage(with: imageUrl, placeholder: UIImage(named: "scribble-placeholder"))
         
         let gradientLayer = CAGradientLayer()
```
Градиент - это такая же вью, как и IBOutlet, лучше создавать её внутри ячейки как свойство)

andrew-kuzn-yndx commented on ImageFeed/ImagesList/ImagesListViewController.swift:
```diff
@@ -94,15 +132,45 @@ extension ImagesListViewController: UITableViewDelegate {
+                cell.setIsLiked(self.photos[indexPath.row].isLiked)
+                UIBlockingProgressHUD.dismiss()
+            case .failure:
+                UIBlockingProgressHUD.dismiss()
```
Дублирование UIBlockingProgressHUD. Его можно вывести за пределы switch, так как его вызывать в любом случае)

andrew-kuzn-yndx commented on ImageFeed/SingleImage/SingleImageViewController.swift:
```diff
@@ -6,45 +6,84 @@
     @IBAction func didTapBackButton(_ sender: Any) {
         dismiss(animated: true, completion: nil)
     }
     @IBAction func didTapShareButton(_ sender: Any) {
```
Стоит проверить весь проект на пропуски private

DarkRavenn committed:
Правки по ревью:
- Поправлен уровень доступа к IBAction
- Убрал дублирование UIBlockingProgressHUD в imageListCellDidTapLike
- Сделал не прямое обращение к аутлетам ImagesListCell
- Вынес Форматирование даты в отдельный extension

DarkRavenn commented:
@andrew-kuzn-yndx глянь плз по возможности по правкам не везде уверен что сделано хорошо.

DarkRavenn committed:
fixed: Неправильное состояние likeButton при запуске приложения
